app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LecturesByRoom.kt

```diff
@@ -0,0 +1,27 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+import java.util.HashMap
+
+data class LecturesByRoom(val lectures: List<Lecture>): HashMap<Int, List<Lecture>>() {
+
+    init{
+        val maxRoomIndex = lectures.map { it.roomIndex }.max() ?: 0
+        val roomCount = maxRoomIndex + 1
+
+        for (roomIndex in 0 until roomCount) {
+            for (lecture in lectures) {
+                if (lecture.roomIndex == roomIndex) {
+                    if (!this.containsKey(roomIndex)) {
+                        this[roomIndex] = mutableListOf()
+                    }
```

Comment:
:lipstick: Can be replaced with `putIfAbsent(roomIndex, mutableListOf())`.

Reply:
Only from API level 24, we are on 14.

Reply:
My bad. I remember that. But Lint did not wake up fast enough to warn me. Kotlin has `getOrPut` - maybe you want to check this. I leave it up to you and close for now.